pkgs/applications/networking/syncthing/default.nix

```diff
@@ -1,57 +1,99 @@
-{ stdenv, lib, fetchFromGitHub, go, procps, removeReferencesTo }:
+{ stdenv, lib, go, procps, removeReferencesTo, fetchFromGitHub }:
 
-stdenv.mkDerivation rec {
-  version = "0.14.44";
-  name = "syncthing-${version}";
+let
+  common =
+    args@{ name, target, installPhase }:
 
-  src = fetchFromGitHub {
-    owner  = "syncthing";
-    repo   = "syncthing";
-    rev    = "v${version}";
-    sha256 = "1gdkx6lbzmdz2hqc9slbq41rwgkxmdisnj0iywx4mppmc2b4v6wh";
-  };
+    stdenv.mkDerivation rec {
+      version = "0.14.44";
+      name = "${args.name}-${version}";
+
+      src = fetchFromGitHub {
+        owner  = "syncthing";
+        repo   = "syncthing";
+        rev    = "v${version}";
+        sha256 = "1gdkx6lbzmdz2hqc9slbq41rwgkxmdisnj0iywx4mppmc2b4v6wh";
+      };
+
+      buildInputs = [ go removeReferencesTo ];
+
+      buildPhase = ''
+        # Syncthing expects that it is checked out in $GOPATH, if that variable is
+        # set.  Since this isn't true when we're fetching source, we can explicitly
+        # unset it and force Syncthing to set up a temporary one for us.
+        env GOPATH= go run build.go -no-upgrade -version v${version} build ${target}
+      '';
+
+      inherit (args) installPhase;
 
-  buildInputs = [ go removeReferencesTo ];
+      preFixup = ''
+        find $out/bin -type f -exec remove-references-to -t ${go} '{}' '+'
+      '';
 
-  buildPhase = ''
-    mkdir -p src/github.com/syncthing
-    ln -s $(pwd) src/github.com/syncthing/syncthing
-    export GOPATH=$(pwd)
+      meta = with lib; {
+        homepage = https://www.syncthing.net/;
+        description = "Open Source Continuous File Synchronization";
+        license = licenses.mpl20;
+        maintainers = with maintainers; [ pshendry joko peterhoeg ];
```

Comment:
You don't want to add yourself?

Reply:
I wasn't sure what the etiquette here is - if you're okay with it, I've added myself.

Reply:
Absolutely! We need all the maintainers we can get.